fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/ArmorModelRegistry.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import net.minecraft.client.render.entity.model.BipedEntityModel;
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.impl.client.item.ArmorModelRegistryImpl;
+
+/**
+ * A class for registering custom armor models for {@link ItemConvertible}, to be provided by a {@link Provider}.
+ *
+ * <p>
+ * This can be used to replace existing vanilla armor models conditionally, however each {@link net.minecraft.item.Item}
+ * instance can only allow one {@link Provider}.
+ * </p>
+ */
+@Environment(EnvType.CLIENT)
+public final class ArmorModelRegistry {
+	private ArmorModelRegistry() {
+	}
+
+	/**
+	 * Registers a provider for custom armor models for an item.
+	 *
+	 * @param provider the provider for the model
+	 * @param items    the items to be registered for
+	 */
+	public static void register(/* @Nullable */ Provider provider, ItemConvertible... items) {
+		register(provider, Arrays.asList(items));
+	}
+
+	/**
+	 * Registers a provider for custom armor models for an item.
+	 *
+	 * @param provider the provider for the model
+	 * @param items    the items to be registered for
+	 */
+	public static void register(/* @Nullable */ Provider provider, Iterable<ItemConvertible> items) {
+		ArmorModelRegistryImpl.register(provider, Objects.requireNonNull(items));
```

Comment:
We usually check preconditions first, then blank line, then implementation.

Reply:
Not documented in conventions. Will add that to doc for future.